src/nvim/log.c

```diff
@@ -100,10 +100,17 @@ static bool log_path_init(void)
   return true;
 }
 
-void log_init(void)
+void log_init(int stage)
 {
-  uv_mutex_init(&mutex);
-  log_path_init();
+  if (stage == 0) {
+    // BEFORE set_init_1, which may try logging! 6f27f5ef91b3 #10183
+    uv_mutex_init(&mutex);
+  } else if (stage == 1) {
+    // AFTER init_homedir ("~", XDG) and set_init_1 (env vars). 22b52dd462e5 #11501
+    log_path_init();
```

Comment:
This is ugly. But hopefully can eliminate the logger mutex after "remote TUI" replaces the thread-based TUI.